# Using a seurat object {#seuratobject}


Load an existing object

Plot some clusters on a umap.

Plot some gene expression on a umap

plot some gene expression by cluster.



Look at a table of cell metadata.

Look at counts data / expression data

Does gene exist?


## Load an existing Seurat object



```{r libloader, results='hide', message=FALSE, warning=FALSE} 
library(tidyverse)
library(Seurat)
```

```{r}
pbmc <- readRDS("data/pbmc_tutorial.rds")
```


## Plotting


```{r}
DimPlot(pbmc, reduction = "umap")
```



```{r}
FeaturePlot(pbmc, features =  "CD14")
```

```{r}
VlnPlot(pbmc, features = 'CD14')
```



## How is the data stored?


Some of the most important information for working with Seurat objects is in the metatdata.
This is cell level information.

```{r}
head(pbmc@meta.data)
```

That doesn't have any gene expression though, that's stored in 






<!-- Taken from MBP material,  -->

#### Discussion: The Seurat Object in R {.challenge}

Lets take a look at the seurat object we have just created in R, `pbmc`

To accomodate the complexity of data arising from a single cell RNA seq experiment, the seurat object keeps this as a container of multiple data tables that are linked.

The functions in seurat can access parts of the data object for analysis and visualisation, we will cover this later on. 

There are a couple of concepts to discuss here.
<details>
<summary>**Class**</summary>

These are essentially data containers in R as a class, and can accessed as a variable in the R environment. 

Classes are pre-defined and can contain multiple data tables and metadata. For Seurat, there are three types. 

* Seurat - the main data class, contains all the data.
* Assay - found within the Seurat object. Depending on the experiment a cell could have data on RNA, ATAC etc measured
* DimReduc - for PCA and UMAP

</details>

<details>
<summary>**Slots**</summary>

Slots are parts within a class that contain specific data. These can be lists, data tables and vectors and can be accessed with conventional R methods.
 
</details>


<details>
<summary>**Data Access**</summary>

Many of the functions in Seurat operate on the data class and slots within them seamlessly. There maybe occasion to access these separately to `hack` them, however this is an advanced analysis method. 

The ways to access the slots can be through methods for the class (functions) or with standard R accessor nomenclature.
</details>

**Examples of accessing a Seurat object**

The `assays` slot in `pbmc` can be accessed with `pbmc@assays`.

The `RNA` assay can be accessed from this with `pbmc@assays$RNA`. 

We often want to access assays, so Seurat nicely gives us a shortcut `pbmc$RNA`. You may sometimes see an alternative notation `pbmc[["RNA"]]`.

In general, slots that are always in an object are accessed with `@` and things that may be different in different data sets are accessed with `$`.

**Have a go**

Use `str` to look at the structure of the Seurat object `pbmc`.

What is in the `meta.data` slot within your Seurat object currently? What type of data is contained here?

Where is our count data within the Seurat object? 

