--- 
title: "scRNAseq Analysis in R with Seurat"
date: 'Compiled: `r format(Sys.Date(), "%B %d, %Y")`'
site: bookdown::bookdown_site
documentclass: book
bibliography: [book.bib, packages.bib]
# url: your book url like https://bookdown.org/yihui/bookdown
# cover-image: path to the social sharing image like images/cover.jpg
description: |
  scRNAseq analysis in R with Seurat workshop material 
link-citations: yes
github-repo: rstudio/bookdown-demo
---

# url: your book url like https://bookdown.org/yihui/bookdown

Placeholder


## Summary
### Recommended Computer Requirements:

<!--chapter:end:index.Rmd-->


# Set up

Placeholder


## Get the workshop material and data
## Package Installation
## Raw Data

<!--chapter:end:01-01-setup.Rmd-->


# M1 Installation

Placeholder


## Instructions That Should Work
## Instructions That Will Fail
## Troubleshooting installation on M1 Mac with R.4.2:

<!--chapter:end:01-02-macos_installation.Rmd-->


# Schedule {#schedule}

Placeholder


### Day 1 {-}
### Day 2 {-}

<!--chapter:end:02-schedual.Rmd-->


# (PART) Seurat Object {-}  

Placeholder


#### Discussion: The Seurat Object in R { - .rational}
## Load an existing Seurat object
## What's in there?
## Plotting
#### Challenge: Plotting { - .challenge}

<!--chapter:end:04-01-seuratobject.Rmd-->


# Load data {#load}

Placeholder


## Setup the Seurat Object
#### Note: What does the data look like? { - .challenge}
### { - }

<!--chapter:end:04-02-load.Rmd-->


# (PART) Single Cell Analysis {-}  

Placeholder


## QC and selecting cells for further analysis
#### Why do we need to do this? {- .rational}
####  {-}
#### Challenge: The meta.data slot in the Seurat object { - .challenge}
####  {-}
#### Challenge: Filter the cells {- .challenge}

<!--chapter:end:04-03-qc.Rmd-->

```{r include=FALSE, cache=FALSE}
# example R options set globally
options(width = 60)

# example chunk options set globally
knitr::opts_chunk$set(
  comment = "#>",
  collapse = TRUE
  )
```
# Normalisation {#norm}


#### Why do we need to do this? {- .rational}
The sequencing depth can be different per cell. This can bias the counts of expression showing higher numbers for more sequenced cells leading to the wrong biological conclusions. To correct this the feature counts are normalized.
                              
####  {-}


After removing unwanted cells from the dataset, the next step is to normalize the data. By default, we employ a global-scaling normalization method "LogNormalize" that normalizes the feature expression measurements for each cell by the total expression, multiplies this by a scale factor (10,000 by default), and log-transforms the result. Normalized values are stored in `pbmc$RNA@data`.

```{r normalize}
pbmc <- NormalizeData(pbmc, normalization.method = "LogNormalize", scale.factor = 1e4)
```
For clarity, in this previous line of code (and in future commands), we provide the default values for certain parameters in the function call. However, this isn't required and the same behavior can be achieved with:

```{r normalize.default, eval = FALSE}
pbmc <- NormalizeData(pbmc)
```









<!--chapter:end:04-04-norm.Rmd-->


# PCAs and UMAPs {#reducedims}

Placeholder


## Identification of highly variable features (feature selection)
#### Why do we need to do this? {- .rational}
####  {-}
## Scaling the data
#### Why do we need to do this? {- .rational}
####  {-}
#### Why do we need to do this? {- .rational}
####  {-}                    
## Perform linear dimensional reduction
## Determine the 'dimensionality' of the dataset
## Run non-linear dimensional reduction (UMAP/tSNE)
#### Challenge: PC genes { - .challenge}
####  {-}
## Save 

<!--chapter:end:04-05-reducedims.Rmd-->


# Clustering {#clustering}

Placeholder


#### Why do we need to do this?  {- .rational}
####  {-}
## Cluster cells
#### Challenge: Try different cluster settings {- .challenge}
####  {-}
## Choosing a cluster resolution

<!--chapter:end:04-06-clustering.Rmd-->


# Cluster Markers {#clustermarkers}

Placeholder


#### Why do we need to do this?  {- .rational}
####  {-}
## Finding differentially expressed features (cluster biomarkers)
## Use makers to label or find a cluster
## Assigning cell type identity to clusters

<!--chapter:end:04-07-clustermarkers.Rmd-->


# (PART) Futher Analysis {-} 

Placeholder



<!--chapter:end:04-08-singleR.Rmd-->


# Differential Expression {#de2}

Placeholder


## Prefiltering
#### Why do we need to do this? {- .rational}
#### {-}
##  Default Wilcox test
## Seurat Negative binomial
## Pseudobulk
#### Discussion {- .challenge}
#### {-}

<!--chapter:end:04-10-de2.Rmd-->


# Cell cycle Assignment {#CellCycle}

Placeholder



<!--chapter:end:04-11-CellCycle.Rmd-->


# Data set integration with Harmony {#Harmony}

Placeholder


### Why do we need to do this? {- .rational}
### {-}

<!--chapter:end:04-12-Harmony.Rmd-->


# (PART) Other resources {-}
# (PART) Other Resources {-} 

Placeholder


### Suggested Further Reading Material
### Useful links arising from the discussion during the previous workshop
#### Data interpretation
#### Data tools and visualisation 
#### Papers
#### Reference data and databases
## Help and fruther Resources
### Seurat Vignettes {-}
### Seurat Cheatsheet {-}
### OSCA {-}
### MBP training Reading list {-}
### Biocommons Single Cell Omics {-}
## Data
### Demo 10X data {-}
### GEO {-}
### Seurat data {-}
## Analysis Tools 
### Cyclone {-}
### Harmony {-}
### SingleR {-}
### Scrublet {-}
### ScVelo {-}
### Monocle {-}
### TidySeurat {-}
## Preprocessing Tools
### Cell Ranger {-}
### STARSolo {-}

<!--chapter:end:05-01-resources.Rmd-->


# Solutions

Placeholder


## Discussion: The Seurat Object in R
## Challenge: The meta.data slot in the Seurat object
## Challenge: Ribosomal gene expression as a QC metric
## Challenge: Labelling Genes of Interest
## Challenge: Try different cluster settings
## Challenge: Reference Based Annotation

<!--chapter:end:06-solutions.Rmd-->

```{r include=FALSE, cache=FALSE}
# example R options set globally
options(width = 60)

# example chunk options set globally
knitr::opts_chunk$set(
  comment = "#>",
  collapse = TRUE
  )
```
# Acknowledgements

This material is mostly based on the [Seurat introductory tutorial.](https://satijalab.org/seurat/articles/pbmc3k_tutorial.html)

It also draws from material for a [workshop provided by QCIF](https://swbioinf.github.io/scRNAseqInR_Doco/index.html) developed by Sarah Williams and Ahmed Mehdi for the Australian Biocommons.

                  
                                                                                                                                                         
                                                                                                                                                         
                                                                                                                                                         
                                                                                                                                                         

<!--chapter:end:07-Acknowledgements.Rmd-->

---
title: "BuildBookdown"
author: "Laura Perlaza-Jimenez"
date: "2023-11-05"
output: html_document
---
```{r include=FALSE, cache=FALSE}
# example R options set globally
options(width = 60)

# example chunk options set globally
knitr::opts_chunk$set(
  comment = "#>",
  collapse = TRUE
  )
```

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## install bookdown


```{r }
install.packages("bookdown")

```

## open project

open the .Rproj file and in the history panel of Rstudio you will see the ```build``` tab. ```Build Book``` from there.

<!--chapter:end:buildbook_instructions.Rmd-->

