<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 12 Differential Expression | scRNAseq Analysis in R with Seurat</title>
<meta name="description" content="There are many different methods for calculating differential expression between groups in scRNAseq data. There are a number of review papers worth consulting on this topic. There is the Seurat...">
<meta name="generator" content="bookdown 0.27 with bs4_book()">
<meta property="og:title" content="Chapter 12 Differential Expression | scRNAseq Analysis in R with Seurat">
<meta property="og:type" content="book">
<meta property="og:description" content="There are many different methods for calculating differential expression between groups in scRNAseq data. There are a number of review papers worth consulting on this topic. There is the Seurat...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 12 Differential Expression | scRNAseq Analysis in R with Seurat">
<meta name="twitter:description" content="There are many different methods for calculating differential expression between groups in scRNAseq data. There are a number of review papers worth consulting on this topic. There is the Seurat...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.3.1/transition.js"></script><script src="libs/bs3compat-0.3.1/tabs.js"></script><script src="libs/bs3compat-0.3.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script><link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="libs/datatables-binding-0.20/datatables.js"></script><link href="libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script><link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">scRNAseq Analysis in R with Seurat</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> About</a></li>
<li><a class="" href="welcome.html"><span class="header-section-number">2</span> Welcome</a></li>
<li><a class="" href="schedule.html"><span class="header-section-number">3</span> Schedule</a></li>
<li class="book-part">Content</li>
<li><a class="" href="seuratobject.html"><span class="header-section-number">4</span> The Seurat object</a></li>
<li class="book-part">Seurat PBMC3k Tutorial</li>
<li><a class="" href="load.html"><span class="header-section-number">5</span> Load data</a></li>
<li><a class="" href="qc.html"><span class="header-section-number">6</span> QC Filtering</a></li>
<li><a class="" href="norm.html"><span class="header-section-number">7</span> Normalisation</a></li>
<li><a class="" href="reducedims.html"><span class="header-section-number">8</span> PCAs and UMAPs</a></li>
<li><a class="" href="clustering.html"><span class="header-section-number">9</span> Clustering</a></li>
<li><a class="" href="cluster-markers.html"><span class="header-section-number">10</span> Cluster Markers</a></li>
<li class="book-part">Futher Analysis</li>
<li><a class="" href="singler.html"><span class="header-section-number">11</span> SingleR</a></li>
<li><a class="active" href="de.html"><span class="header-section-number">12</span> Differential Expression</a></li>
<li class="book-part">End</li>
<li><a class="" href="finishing-up.html"><span class="header-section-number">13</span> Finishing up</a></li>
<li><a class="" href="resources.html"><span class="header-section-number">14</span> Resources</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/rstudio/bookdown-demo">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="de" class="section level1" number="12">
<h1>
<span class="header-section-number">12</span> Differential Expression<a class="anchor" aria-label="anchor" href="#de"><i class="fas fa-link"></i></a>
</h1>
<p>There are many different methods for calculating differential expression between groups in scRNAseq data. There are a number of review papers worth consulting on this topic.</p>
<p>There is the <a href="https://satijalab.org/seurat/archive/v3.1/de_vignette.html">Seurat differential expression Vignette</a> which walks through the variety implemented in Seurat.</p>
<p>The example below quickly looks at 3 different methods on a test dataset. It is by no means a complete guide!</p>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat">Seurat</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; ── Attaching packages ─────────────────── tidyverse 1.3.1 ──</span></span>
<span><span class="co">#&gt; ✔ tibble  3.1.6     ✔ purrr   0.3.4</span></span>
<span><span class="co">#&gt; ✔ tidyr   1.1.4     ✔ stringr 1.4.0</span></span>
<span><span class="co">#&gt; ✔ readr   2.1.1     ✔ forcats 0.5.1</span></span>
<span><span class="co">#&gt; ── Conflicts ────────────────────── tidyverse_conflicts() ──</span></span>
<span><span class="co">#&gt; ✖ IRanges::collapse()      masks dplyr::collapse()</span></span>
<span><span class="co">#&gt; ✖ Biobase::combine()       masks BiocGenerics::combine(), dplyr::combine()</span></span>
<span><span class="co">#&gt; ✖ matrixStats::count()     masks dplyr::count()</span></span>
<span><span class="co">#&gt; ✖ IRanges::desc()          masks dplyr::desc()</span></span>
<span><span class="co">#&gt; ✖ tidyr::expand()          masks S4Vectors::expand()</span></span>
<span><span class="co">#&gt; ✖ dplyr::filter()          masks stats::filter()</span></span>
<span><span class="co">#&gt; ✖ S4Vectors::first()       masks dplyr::first()</span></span>
<span><span class="co">#&gt; ✖ dplyr::lag()             masks stats::lag()</span></span>
<span><span class="co">#&gt; ✖ BiocGenerics::Position() masks ggplot2::Position(), base::Position()</span></span>
<span><span class="co">#&gt; ✖ purrr::reduce()          masks GenomicRanges::reduce(), IRanges::reduce()</span></span>
<span><span class="co">#&gt; ✖ S4Vectors::rename()      masks dplyr::rename()</span></span>
<span><span class="co">#&gt; ✖ IRanges::slice()         masks dplyr::slice()</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com">patchwork</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/rstudio/DT">DT</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'DT'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:SeuratObject':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     JS</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:Seurat':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     JS</span></span>
<span></span>
<span><span class="va">condition_lookup</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'MantonBM1'</span><span class="op">=</span><span class="st">"fake_treat"</span>,   <span class="st">'MantonBM2'</span><span class="op">=</span><span class="st">"fake_treat"</span>,   <span class="st">'MantonBM3'</span><span class="op">=</span><span class="st">"fake_treat"</span>,   <span class="st">'MantonBM4'</span><span class="op">=</span><span class="st">"fake_treat"</span>,</span>
<span>    <span class="st">'MantonBM5'</span><span class="op">=</span><span class="st">"fake_control"</span>, <span class="st">'MantonBM6'</span><span class="op">=</span><span class="st">"fake_control"</span>, <span class="st">'MantonBM7'</span><span class="op">=</span><span class="st">"fake_control"</span>, <span class="st">'MantonBM8'</span><span class="op">=</span><span class="st">"fake_control"</span><span class="op">)</span></span></code></pre></div>
<div id="prepared-data" class="section level2" number="12.1">
<h2>
<span class="header-section-number">12.1</span> Prepared data<a class="anchor" aria-label="anchor" href="#prepared-data"><i class="fas fa-link"></i></a>
</h2>
<p>No need to run this - load up the pre prepared Seurat object.</p>
<p>This code downloads and modifies the hcabm40k Human Cell Atlas Bone ICA marrow dataset to something small to test differential expression with.
This dataset has data from 8 individuals (pbmc3k is all from one doner).</p>
<ol style="list-style-type: decimal">
<li>Downloads the hcabm40k dataset and subsamples it to just 4000 cells.</li>
<li>Assigns the 8 individuals to ‘fake_treat’ or ‘fake_control’ groups. There should effectively be no differential expression between them.</li>
<li>Take the first 50 genes and changes expression in the ‘fake treat’ group - multiply counts by 4 and add 1 to a subset. Now we can expect these genes to be ‘differentially expressed’</li>
</ol>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">################################################################################</span></span>
<span><span class="co"># Prepare a datasets</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://www.satijalab.org/seurat">SeuratData</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"hcabm40k"</span><span class="op">)</span> </span>
<span></span>
<span><span class="co"># grab 4000 random cells</span></span>
<span><span class="va">keep_cells</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">hcabm40k</span><span class="op">)</span>, size<span class="op">=</span><span class="fl">4000</span><span class="op">)</span></span>
<span>  </span>
<span><span class="co"># Subsample down to a toy sized dataset</span></span>
<span><span class="va">hcabm40k.mini</span> <span class="op">&lt;-</span> <span class="va">hcabm40k</span><span class="op">[</span>,<span class="va">keep_cells</span><span class="op">]</span> </span>
<span><span class="co"># clear large object from memory</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span><span class="va">hcabm40k</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add a group of contidion vs control to test. Not real</span></span>
<span><span class="co"># This is the first 4 of 8 bio replicates</span></span>
<span><span class="va">hcabm40k.mini</span><span class="op">$</span><span class="va">condition</span> <span class="op">&lt;-</span> <span class="va">condition_lookup</span><span class="op">[</span><span class="va">hcabm40k.mini</span><span class="op">@</span><span class="va">meta.data</span><span class="op">$</span><span class="va">orig.ident</span><span class="op">]</span></span>
<span></span>
<span></span>
<span></span>
<span><span class="co"># ADDING SOME FAKE VARIATION!</span></span>
<span><span class="co"># Mmultipling first 40 genes of the the 'fake treat' group by 4 - so we expect these to be DE.</span></span>
<span><span class="co"># In reality, these are simply healthy replicates and shouldn't be significantly different</span></span>
<span><span class="va">alter_cells</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">condition_lookup</span><span class="op">[</span><span class="va">hcabm40k.mini</span><span class="op">$</span><span class="va">orig.ident</span><span class="op">]</span> <span class="op">==</span> <span class="st">"fake_treat"</span><span class="op">)</span></span>
<span><span class="va">alter_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">hcabm40k.mini</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">RNA</span><span class="op">@</span><span class="va">counts</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">50</span><span class="op">]</span> </span>
<span><span class="va">hcabm40k.mini</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">RNA</span><span class="op">@</span><span class="va">counts</span><span class="op">[</span><span class="va">alter_genes</span>, <span class="va">alter_cells</span> <span class="op">]</span> <span class="op">&lt;-</span> <span class="va">hcabm40k.mini</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">RNA</span><span class="op">@</span><span class="va">counts</span><span class="op">[</span><span class="va">alter_genes</span>, <span class="va">alter_cells</span>  <span class="op">]</span><span class="op">*</span> <span class="fl">4</span></span>
<span><span class="co"># Plus - add 1 to some of them too (2*0 is 0)</span></span>
<span><span class="va">alter_cells2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span><span class="op">(</span><span class="va">alter_cells</span>, <span class="fl">500</span>, replace <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="co"># add 1 to x of 1979 cells.</span></span>
<span><span class="va">hcabm40k.mini</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">RNA</span><span class="op">@</span><span class="va">counts</span><span class="op">[</span><span class="va">alter_genes</span>, <span class="va">alter_cells2</span> <span class="op">]</span> <span class="op">&lt;-</span> <span class="va">hcabm40k.mini</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">RNA</span><span class="op">@</span><span class="va">counts</span><span class="op">[</span><span class="va">alter_genes</span>, <span class="va">alter_cells2</span>  <span class="op">]</span><span class="op">*</span> <span class="op">+</span><span class="fl">1</span></span>
<span></span>
<span></span>
<span><span class="co"># there's gene level meta data (meta.features) for each assay. </span></span>
<span><span class="co"># Recodring which values we changed.</span></span>
<span><span class="va">hcabm40k.mini</span><span class="op">[[</span><span class="st">'RNA'</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">'is_altered'</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">hcabm40k.mini</span><span class="op">[[</span><span class="st">'RNA'</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">alter_genes</span></span>
<span></span>
<span></span>
<span></span>
<span><span class="co">#-------------------------------------------------------------------------------</span></span>
<span><span class="co">#  Run thorough a basic processing, with no QC</span></span>
<span></span>
<span><span class="va">hcabm40k.mini</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html">NormalizeData</a></span><span class="op">(</span><span class="va">hcabm40k.mini</span>, normalization.method <span class="op">=</span> <span class="st">"LogNormalize"</span>, scale.factor <span class="op">=</span> <span class="fl">10000</span><span class="op">)</span></span>
<span><span class="va">hcabm40k.mini</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindVariableFeatures.html">FindVariableFeatures</a></span><span class="op">(</span><span class="va">hcabm40k.mini</span>, selection.method <span class="op">=</span> <span class="st">"vst"</span>, nfeatures <span class="op">=</span> <span class="fl">2000</span><span class="op">)</span></span>
<span><span class="va">all.genes</span>     <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">hcabm40k.mini</span><span class="op">)</span></span>
<span><span class="va">hcabm40k.mini</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/ScaleData.html">ScaleData</a></span><span class="op">(</span><span class="va">hcabm40k.mini</span>, features <span class="op">=</span> <span class="va">all.genes</span><span class="op">)</span></span>
<span><span class="va">hcabm40k.mini</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunPCA.html">RunPCA</a></span><span class="op">(</span><span class="va">hcabm40k.mini</span>, features <span class="op">=</span> <span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/VariableFeatures.html">VariableFeatures</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">hcabm40k.mini</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">hcabm40k.mini</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunUMAP.html">RunUMAP</a></span><span class="op">(</span><span class="va">hcabm40k.mini</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">hcabm40k.mini</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindNeighbors.html">FindNeighbors</a></span><span class="op">(</span><span class="va">hcabm40k.mini</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">hcabm40k.mini</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindClusters.html">FindClusters</a></span><span class="op">(</span><span class="va">hcabm40k.mini</span>, resolution <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span> <span class="co"># Low res</span></span>
<span></span>
<span></span>
<span><span class="co">#------------------------------------------------------------------------------</span></span>
<span><span class="co"># save</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">saveRDS</a></span><span class="op">(</span><span class="va">hcabm40k.mini</span>, file<span class="op">=</span><span class="st">"data/hcabm40k_mini_withfakevariation.rds"</span><span class="op">)</span></span></code></pre></div>
</div>
<div id="load-data" class="section level2" number="12.2">
<h2>
<span class="header-section-number">12.2</span> Load Data<a class="anchor" aria-label="anchor" href="#load-data"><i class="fas fa-link"></i></a>
</h2>
<p>Load up the small, edited subset of data.</p>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hcabm40k.mini</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span>file<span class="op">=</span><span class="st">"data/hcabm40k_mini_withfakevariation.rds"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span> <span class="va">hcabm40k.mini</span><span class="op">$</span><span class="va">orig.ident</span>, <span class="va">hcabm40k.mini</span><span class="op">$</span><span class="va">condition</span><span class="op">)</span></span>
<span><span class="co">#&gt;            </span></span>
<span><span class="co">#&gt;             fake_control fake_treat</span></span>
<span><span class="co">#&gt;   MantonBM1            0        502</span></span>
<span><span class="co">#&gt;   MantonBM2            0        498</span></span>
<span><span class="co">#&gt;   MantonBM3            0        481</span></span>
<span><span class="co">#&gt;   MantonBM4            0        524</span></span>
<span><span class="co">#&gt;   MantonBM5          467          0</span></span>
<span><span class="co">#&gt;   MantonBM6          522          0</span></span>
<span><span class="co">#&gt;   MantonBM7          476          0</span></span>
<span><span class="co">#&gt;   MantonBM8          530          0</span></span></code></pre></div>
<p>The first 50 genes that were altered to be higher in the ‘fake_treat’ group.</p>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">altered_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">hcabm40k.mini</span><span class="op">[[</span><span class="st">'RNA'</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">50</span><span class="op">]</span></span>
<span><span class="va">altered_genes</span></span>
<span><span class="co">#&gt;  [1] "RP11-34P13.7"  "FO538757.2"    "AP006222.2"   </span></span>
<span><span class="co">#&gt;  [4] "RP4-669L17.10" "RP5-857K21.4"  "RP11-206L10.9"</span></span>
<span><span class="co">#&gt;  [7] "LINC00115"     "FAM41C"        "SAMD11"       </span></span>
<span><span class="co">#&gt; [10] "NOC2L"         "KLHL17"        "PLEKHN1"      </span></span>
<span><span class="co">#&gt; [13] "HES4"          "ISG15"         "AGRN"         </span></span>
<span><span class="co">#&gt; [16] "C1orf159"      "TTLL10"        "TNFRSF18"     </span></span>
<span><span class="co">#&gt; [19] "TNFRSF4"       "SDF4"          "B3GALT6"      </span></span>
<span><span class="co">#&gt; [22] "FAM132A"       "RP5-902P8.12"  "UBE2J2"       </span></span>
<span><span class="co">#&gt; [25] "RP5-902P8.10"  "SCNN1D"        "ACAP3"        </span></span>
<span><span class="co">#&gt; [28] "PUSL1"         "CPSF3L"        "CPTP"         </span></span>
<span><span class="co">#&gt; [31] "TAS1R3"        "DVL1"          "MXRA8"        </span></span>
<span><span class="co">#&gt; [34] "AURKAIP1"      "CCNL2"         "MRPL20"       </span></span>
<span><span class="co">#&gt; [37] "RP4-758J18.13" "ANKRD65"       "VWA1"         </span></span>
<span><span class="co">#&gt; [40] "ATAD3C"        "ATAD3B"        "ATAD3A"       </span></span>
<span><span class="co">#&gt; [43] "SSU72"         "RP5-832C2.5"   "C1orf233"     </span></span>
<span><span class="co">#&gt; [46] "RP11-345P4.9"  "MIB2"          "MMP23B"       </span></span>
<span><span class="co">#&gt; [49] "CDK11B"        "RP11-345P4.10"</span></span></code></pre></div>
</div>
<div id="prefiltering" class="section level2" number="12.3">
<h2>
<span class="header-section-number">12.3</span> Prefiltering<a class="anchor" aria-label="anchor" href="#prefiltering"><i class="fas fa-link"></i></a>
</h2>
<p>When doing differential expression, you generally ignore genes with low exprssion.
If expression is below a certain level, it will be almost impossible to see differential expression.</p>
<p>In single cell datasets, there are many genes like this. Aggressively filtering here to make our dataset smaller so it runs quicker.</p>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">total_per_gene</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowSums</a></span><span class="op">(</span><span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/AssayData.html">GetAssayData</a></span><span class="op">(</span><span class="va">hcabm40k.mini</span>, <span class="st">'counts'</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="va">total_per_gene</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="scRNAseqInR_Doco_files/figure-html/unnamed-chunk-40-1.png" width="672"></div>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hcabm40k.mini</span> <span class="op">&lt;-</span> <span class="va">hcabm40k.mini</span><span class="op">[</span><span class="va">total_per_gene</span> <span class="op">&gt;=</span> <span class="fl">200</span>, <span class="op">]</span> </span>
<span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html">DimPlot</a></span><span class="op">(</span><span class="va">hcabm40k.mini</span>, reduction <span class="op">=</span> <span class="st">'umap'</span>, group.by <span class="op">=</span> <span class="st">'seurat_clusters'</span><span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html">DimPlot</a></span><span class="op">(</span><span class="va">hcabm40k.mini</span>, reduction <span class="op">=</span> <span class="st">'umap'</span>, group.by <span class="op">=</span> <span class="st">'orig.ident'</span><span class="op">)</span></span>
<span><span class="va">p1</span><span class="op">+</span><span class="va">p2</span></span></code></pre></div>
<div class="inline-figure"><img src="scRNAseqInR_Doco_files/figure-html/unnamed-chunk-40-2.png" width="672"></div>
</div>
<div id="default-wilcox-test" class="section level2" number="12.4">
<h2>
<span class="header-section-number">12.4</span> Default Wilcox test<a class="anchor" aria-label="anchor" href="#default-wilcox-test"><i class="fas fa-link"></i></a>
</h2>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hcabm40k.mini.c0</span> <span class="op">&lt;-</span> <span class="va">hcabm40k.mini</span><span class="op">[</span>, <span class="va">hcabm40k.mini</span><span class="op">$</span><span class="va">seurat_clusters</span> <span class="op">==</span> <span class="fl">0</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Change Ident to condition</span></span>
<span><span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/Idents.html">Idents</a></span><span class="op">(</span><span class="va">hcabm40k.mini.c0</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">hcabm40k.mini.c0</span><span class="op">$</span><span class="va">condition</span></span>
<span></span>
<span><span class="co"># default, wilcox test</span></span>
<span><span class="va">de_result_wilcox</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindMarkers.html">FindMarkers</a></span><span class="op">(</span><span class="va">hcabm40k.mini.c0</span>, </span>
<span>            ident.1 <span class="op">=</span> <span class="st">'fake_treat'</span>,</span>
<span>            ident.2 <span class="op">=</span> <span class="st">'fake_control'</span>,</span>
<span>            logfc.threshold <span class="op">=</span> <span class="fl">0</span>, <span class="co"># Give me ALL results</span></span>
<span>            min.pct <span class="op">=</span> <span class="fl">0</span>,</span>
<span>            max.cells.per.ident <span class="op">=</span> <span class="fl">400</span>  <span class="co"># Downsample cells - purely for speed!</span></span>
<span>            <span class="op">)</span></span>
<span></span>
<span><span class="co"># the  'DE' genes</span></span>
<span><span class="va">de_result_wilcox</span><span class="op">$</span><span class="va">was_altered</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">de_result_wilcox</span> <span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">altered_genes</span></span>
<span></span>
<span><span class="co"># Add average expression for plotting</span></span>
<span><span class="va">de_result_wilcox</span><span class="op">$</span><span class="va">AveExpr</span><span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowMeans</a></span><span class="op">(</span><span class="va">hcabm40k.mini.c0</span><span class="op">[[</span><span class="st">"RNA"</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">de_result_wilcox</span><span class="op">)</span>,<span class="op">]</span><span class="op">)</span></span></code></pre></div>
<p>Look at the significantly differentially expressed genes.</p>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">DT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html">datatable</a></span><span class="op">(</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">de_result_wilcox</span>, <span class="va">p_val_adj</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div id="htmlwidget-0b89372b7b7816450716" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-0b89372b7b7816450716">{"x":{"filter":"none","vertical":false,"data":[["RPS4Y1","RPS9","HLA-B","RPS4X","MT-CO1","IFITM1","CRIP1","CTSW","RPS6","FTH1","RNASET2","JUN","MIF","IGKC","AC090498.1","RPS3A","DUSP1","FOS","SRGN","RPL35A","EIF1","TOMM7","CDKN2D","RPL34","RPL7","RPL21","SOCS3","HCST","RPL13","RPS24","NUCB2","CD8A","MPZL3","PPDPF","CD8B","RPS29"],[1.11275775347644e-35,1.33365368791274e-18,7.42581758603555e-17,9.1521611822317e-16,3.33418988835378e-10,4.70831517862278e-10,6.560853473387e-10,1.51425515837624e-09,1.61861859359807e-09,1.87210898878251e-09,4.4154558393769e-09,6.18146146358439e-09,1.03015385861995e-08,1.43537066320284e-08,1.46859871989682e-08,1.72230440614433e-08,4.89236893572775e-08,6.2249337173307e-08,9.47042573195458e-08,1.52134450063921e-07,2.47351780640944e-07,4.7093696149744e-07,5.68284687700273e-07,6.61223574890475e-07,1.03881159637724e-06,1.22121816235969e-06,1.42831788992206e-06,1.520261665577e-06,2.41618293476252e-06,2.42527460220064e-06,3.78374931743423e-06,6.18090936472663e-06,6.42638143364466e-06,7.21129627693999e-06,7.40836070749986e-06,7.95255540577907e-06],[1.39363833201555,-0.310957294801806,0.467365978803999,-0.299449642390523,0.186224299116096,0.633081398515384,0.797378369504039,-0.699011811536696,-0.150377931286037,0.468272172246626,0.443394157361129,-0.484292394277563,0.448380258317102,0.45450004873331,0.427924190420527,0.139173128279262,-0.642040436461872,-0.275896788169862,0.498853852904095,-0.107385222662039,0.247331806379365,0.335271022255575,0.350778961589261,-0.129119812853113,-0.137849424427826,-0.136703067599919,0.472566782091844,-0.327196980897592,-0.09585045177145,-0.198620877642905,-0.469999545056883,-0.411437057298089,0.343309945195044,0.384653287935004,-0.737589816331237,0.114063177150514],[0.642,0.999,0.993,1,0.999,0.334,0.334,0.137,1,0.985,0.459,0.626,0.447,0.35,0.662,1,0.461,0.447,0.644,1,0.99,0.967,0.161,1,0.999,1,0.282,0.593,1,1,0.177,0.098,0.118,0.793,0.133,1],[0.166,0.999,0.975,0.999,0.997,0.172,0.141,0.303,1,0.979,0.294,0.764,0.281,0.175,0.518,0.997,0.589,0.55,0.49,1,0.99,0.932,0.077,1,1,1,0.143,0.666,1,0.997,0.293,0.193,0.044,0.675,0.273,1],[6.42061223755905e-32,7.69518177925649e-15,4.28469674714251e-13,5.28079700214769e-12,1.92382756558013e-06,2.71669785806534e-06,3.7856124541443e-06,8.73725226383091e-06,9.33942928506087e-06,1.08020688652751e-05,2.54771801932047e-05,3.56670326448819e-05,5.94398776423713e-05,8.28208872668041e-05,8.47381461380466e-05,9.93769642345277e-05,0.000282289687591491,0.000359178675489982,0.000546443564733779,0.000877815776868826,0.00142721977429825,0.00271730626784023,0.00327900264803058,0.00381526002711804,0.00599394291109665,0.00704642879681538,0.00824139422485027,0.0087719098103793,0.0139413755335797,0.0139938344546977,0.0218322335615955,0.0356638470344727,0.0370802208721297,0.0416091795179438,0.0427462412822742,0.0458862446913452],[false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false],[0.920239837412564,3.57627437205978,3.05767211917926,4.14643463546018,3.64966916183542,0.403307356708525,0.385040593245245,0.384343645345626,4.67127099222135,3.08928966005543,0.620630570406162,1.68040903247711,0.59318925095129,0.419226203189961,1.08455001096358,4.30936645053111,1.13595414060151,1.08061354385778,1.12996432380591,4.1170374077667,3.18725075412962,2.46034185326326,0.177531552422338,4.93694867063007,4.02095946051273,4.92825576691911,0.348445777705556,1.21598347006118,4.88042605222888,3.56275595313222,0.393248200046754,0.258245480049201,0.124647303099063,1.4475434376951,0.401509209100616,4.7015768268751]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>p_val<\/th>\n      <th>avg_log2FC<\/th>\n      <th>pct.1<\/th>\n      <th>pct.2<\/th>\n      <th>p_val_adj<\/th>\n      <th>was_altered<\/th>\n      <th>AveExpr<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,7]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script><p>And the stats for the altered genes;</p>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">DT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html">datatable</a></span><span class="op">(</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">de_result_wilcox</span>, <span class="va">was_altered</span> <span class="op">==</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div id="htmlwidget-8c5dfa6aec61870bb2d5" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-8c5dfa6aec61870bb2d5">{"x":{"filter":"none","vertical":false,"data":[["SSU72","FO538757.2","AURKAIP1","TNFRSF4","ISG15","UBE2J2","CPTP","NOC2L","RP11-345P4.9","ATAD3B","MIB2","ATAD3A","MRPL20","CDK11B","CCNL2","RP11-206L10.9","SDF4","TNFRSF18","AP006222.2","PUSL1","ACAP3","CPSF3L"],[9.65527448754897e-05,0.00120022110300729,0.0221115958961199,0.0660104105578719,0.0821653451824586,0.092834477245298,0.112044115803745,0.134463549064201,0.141111193933961,0.155121448340074,0.265139586368059,0.295248959505662,0.296760415284022,0.342497039600304,0.420548184001565,0.437036971236827,0.652323324886553,0.688764135291486,0.779560031641426,0.823256997058824,0.887775095257728,0.9748765979136],[1.36281990144545,1.20415411290446,1.16643695313443,0.556081682336015,1.19360510657403,0.698385023044602,0.132963557687957,0.831077063827109,0.352680400094964,0.353760005578406,0.837589345237074,0.300268469608446,1.06804937111771,0.690249160634361,0.743717253862625,0.377885338717425,0.806467251598397,0.199601305582793,0.0662005199658423,0.239378825139914,0.325600056356764,0.728706210074751],[0.338,0.103,0.219,0.048,0.162,0.085,0.013,0.092,0.031,0.022,0.089,0.019,0.176,0.08,0.07,0.028,0.103,0.019,0.007,0.018,0.027,0.073],[0.338,0.043,0.207,0.026,0.161,0.101,0.024,0.07,0.023,0.01,0.073,0.019,0.177,0.08,0.062,0.018,0.114,0.016,0.013,0.02,0.031,0.063],[0.557109337931576,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true],[0.736099896964758,0.168601118893333,0.451910936858197,0.0792362721207836,0.351715361671646,0.189799770536085,0.0357293154837701,0.169328127818687,0.0559101555020552,0.0363747254730834,0.170400468330629,0.0397598756488997,0.374288968299508,0.16470705567725,0.136177638873421,0.0490452684660153,0.225995218704633,0.0377898022419026,0.0192200976710924,0.0401112076831192,0.058634481211794,0.140417806062689]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>p_val<\/th>\n      <th>avg_log2FC<\/th>\n      <th>pct.1<\/th>\n      <th>pct.2<\/th>\n      <th>p_val_adj<\/th>\n      <th>was_altered<\/th>\n      <th>AveExpr<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,7]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script><p>Here the blue crosses represent the altered genes, and the red are the differentially expressed.</p>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">de_result_wilcox</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">AveExpr</span>, y<span class="op">=</span><span class="va">avg_log2FC</span>, col<span class="op">=</span><span class="va">p_val_adj</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span>data<span class="op">=</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">de_result_wilcox</span>, <span class="va">was_altered</span><span class="op">)</span>, pch<span class="op">=</span><span class="fl">3</span>, col<span class="op">=</span><span class="st">'blue'</span>, size<span class="op">=</span><span class="fl">5</span> <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_colour_manual</a></span><span class="op">(</span>values<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'TRUE'</span><span class="op">=</span><span class="st">"red"</span>,<span class="st">'FALSE'</span><span class="op">=</span><span class="st">"black"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Wilcox Test"</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="scRNAseqInR_Doco_files/figure-html/unnamed-chunk-44-1.png" width="672"></div>
</div>
<div id="seurat-negative-binomial" class="section level2" number="12.5">
<h2>
<span class="header-section-number">12.5</span> Seurat Negative binomial<a class="anchor" aria-label="anchor" href="#seurat-negative-binomial"><i class="fas fa-link"></i></a>
</h2>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">de_result_negbinom</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindMarkers.html">FindMarkers</a></span><span class="op">(</span><span class="va">hcabm40k.mini.c0</span>, </span>
<span>            ident.1 <span class="op">=</span> <span class="st">'fake_treat'</span>,</span>
<span>            ident.2 <span class="op">=</span> <span class="st">'fake_control'</span>,</span>
<span>            logfc.threshold <span class="op">=</span> <span class="fl">0</span>, <span class="co"># Give me ALL results</span></span>
<span>            min.pct <span class="op">=</span> <span class="fl">0</span>,</span>
<span>            test.use<span class="op">=</span><span class="st">"negbinom"</span>, <span class="co"># Choose a different test.</span></span>
<span>            max.cells.per.ident <span class="op">=</span> <span class="fl">300</span>  <span class="co"># Downsample cells - purely for speed!</span></span>
<span>            <span class="op">)</span></span>
<span><span class="co"># the  'DE' genes</span></span>
<span><span class="va">de_result_negbinom</span><span class="op">$</span><span class="va">was_altered</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">de_result_negbinom</span>  <span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">altered_genes</span></span>
<span><span class="co"># Add average expression for plotting</span></span>
<span><span class="va">de_result_negbinom</span><span class="op">$</span><span class="va">AveExpr</span><span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowMeans</a></span><span class="op">(</span><span class="va">hcabm40k.mini.c0</span><span class="op">[[</span><span class="st">"RNA"</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">de_result_negbinom</span><span class="op">)</span>,<span class="op">]</span><span class="op">)</span></span></code></pre></div>
<p>Look at the significantly differentially expressed genes:</p>
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">DT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html">datatable</a></span><span class="op">(</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">de_result_negbinom</span>, <span class="va">p_val_adj</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div id="htmlwidget-96338203b4d4cbfa578c" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-96338203b4d4cbfa578c">{"x":{"filter":"none","vertical":false,"data":[["HLA-B","RPS4Y1","FTH1","MT-CO1","SSU72","IFITM1","AC090498.1","EIF1","MRPL20","AURKAIP1","CTSW","RORA","RPS3A","IGKC","ISG15","YBX1","RPS29","TOMM7","CRIP1","MIF","PIK3R1","CD8B","PPDPF","STAT3","RPS9","DDX6","FO538757.2","MPZL3","RNASET2","SOCS3","CDKN2D"],[2.04750493689275e-16,9.22031870074104e-16,4.12212672907565e-14,2.2363676793148e-13,3.83059210186881e-13,7.98229240149372e-10,9.0994022627282e-10,1.02172457624812e-09,1.02833089682338e-09,3.83018451960039e-09,7.09945477965296e-09,8.59778642747145e-09,9.91956720522607e-09,2.51118246771585e-08,2.61158849435189e-08,1.14919858512495e-07,2.59118632115393e-07,5.50610803070526e-07,7.20476716866656e-07,7.58068630215243e-07,1.67455170647446e-06,1.78681422181742e-06,2.49360146670056e-06,2.91248515509891e-06,3.062763808356e-06,3.15938852184356e-06,4.27091931006637e-06,4.7341367112222e-06,4.81134942698407e-06,5.79598853620104e-06,5.95972191795326e-06],[0.509307519874486,0.940996221493534,0.559955137160247,0.305832992369344,0.88086768037823,0.335173880809597,0.326193607189878,0.354141402962262,0.54966680021533,0.66476829565517,-0.301423981315451,0.209480420602247,0.248398915841652,0.268594641612714,0.734038986744665,0.271722422953895,0.204779645230391,0.356545116055121,0.407002333469917,0.262515515239564,0.284965227450062,-0.359568451905521,0.334696868192069,0.170994125771409,-0.204565492078604,0.143045339261938,0.469343431621565,0.125785777703207,0.274036473254836,0.252568788070077,0.145721241208925],[0.993,0.642,0.985,0.999,0.338,0.334,0.662,0.99,0.176,0.219,0.137,0.317,1,0.35,0.162,0.811,1,0.967,0.334,0.447,0.38,0.133,0.793,0.364,0.999,0.32,0.103,0.118,0.459,0.282,0.161],[0.975,0.166,0.979,0.997,0.338,0.172,0.518,0.99,0.177,0.207,0.303,0.2,0.997,0.175,0.161,0.74,1,0.932,0.141,0.281,0.244,0.273,0.675,0.255,0.999,0.227,0.043,0.044,0.294,0.143,0.077],[1.18141034858712e-12,5.32012389032758e-12,2.37846712267665e-10,1.29038415096464e-09,2.21025164277831e-09,4.60578271566188e-06,5.25035510559417e-06,5.89535080495163e-06,5.93346927467092e-06,2.21001646780943e-05,4.09638540785976e-05,4.96092276865102e-05,5.72359027741544e-05,0.000144895228387204,0.000150688656124104,0.000663087583617096,0.00149511450730582,0.00317702433371694,0.0041571506563206,0.00437405599634195,0.00966216334635761,0.0103099180598865,0.0143880804628622,0.0168050393449207,0.0176721471742141,0.0182296717710373,0.024643204419083,0.0273159688237521,0.0277614861936981,0.03344285385388,0.0343875954665903],[false,false,false,false,true,false,false,false,true,true,false,false,false,false,true,false,false,false,false,false,false,false,false,false,false,false,true,false,false,false,false],[3.05767211917926,0.920239837412564,3.08928966005543,3.64966916183542,0.736099896964758,0.403307356708525,1.08455001096358,3.18725075412962,0.374288968299508,0.451910936858197,0.384343645345626,0.421328271981957,4.30936645053111,0.419226203189961,0.351715361671646,1.58583684952854,4.7015768268751,2.46034185326326,0.385040593245245,0.59318925095129,0.54477934393953,0.401509209100616,1.4475434376951,0.496669344823902,3.57627437205978,0.444782453367723,0.168601118893333,0.124647303099063,0.620630570406162,0.348445777705556,0.177531552422338]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>p_val<\/th>\n      <th>avg_log2FC<\/th>\n      <th>pct.1<\/th>\n      <th>pct.2<\/th>\n      <th>p_val_adj<\/th>\n      <th>was_altered<\/th>\n      <th>AveExpr<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,7]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script><p>And the stats for the altered genes:</p>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">DT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html">datatable</a></span><span class="op">(</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">de_result_negbinom</span>, <span class="va">was_altered</span> <span class="op">==</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div id="htmlwidget-fdc6da1537b97536e25f" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-fdc6da1537b97536e25f">{"x":{"filter":"none","vertical":false,"data":[["SSU72","MRPL20","AURKAIP1","ISG15","FO538757.2","MIB2","SDF4","NOC2L","CCNL2","CPSF3L","CDK11B","UBE2J2","TNFRSF4","ATAD3B","RP11-345P4.9","RP11-206L10.9","PUSL1","ATAD3A","AP006222.2","ACAP3","CPTP","TNFRSF18"],[3.83059210186881e-13,1.02833089682338e-09,3.83018451960039e-09,2.61158849435189e-08,4.27091931006637e-06,1.98100619005395e-05,6.9437319734247e-05,0.000108920303753501,0.000139832308101614,0.000156947149671535,0.000311090267948707,0.00373276597626871,0.00832138978798791,0.0115876040929247,0.0120833368092726,0.0127370572633494,0.0374542604113871,0.0626999097710347,0.203995121366936,0.211134937855113,0.360323402034061,0.572421701709994],[0.88086768037823,0.54966680021533,0.66476829565517,0.734038986744665,0.469343431621565,0.379970025596093,0.358566030665003,0.35397478431421,0.290177041772783,0.296554275933831,0.311821092645369,0.29944875195735,0.2234004281158,0.109131569116974,0.134363156584411,0.13734666408575,0.0709677075975867,0.08074939707731,0.0243615362037566,0.110181685354898,0.0413658200344243,0.0771959592195554],[0.338,0.176,0.219,0.162,0.103,0.089,0.103,0.092,0.07,0.073,0.08,0.085,0.048,0.022,0.031,0.028,0.018,0.019,0.007,0.027,0.013,0.019],[0.338,0.177,0.207,0.161,0.043,0.073,0.114,0.07,0.062,0.063,0.08,0.101,0.026,0.01,0.023,0.018,0.02,0.019,0.013,0.031,0.024,0.016],[2.21025164277831e-09,5.93346927467092e-06,2.21001646780943e-05,0.000150688656124104,0.024643204419083,0.114304057166113,0.400653334866605,0.628470152657702,0.806832417746312,0.905585053604754,1,1,1,1,1,1,1,1,1,1,1,1],[true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true],[0.736099896964758,0.374288968299508,0.451910936858197,0.351715361671646,0.168601118893333,0.170400468330629,0.225995218704633,0.169328127818687,0.136177638873421,0.140417806062689,0.16470705567725,0.189799770536085,0.0792362721207836,0.0363747254730834,0.0559101555020552,0.0490452684660153,0.0401112076831192,0.0397598756488997,0.0192200976710924,0.058634481211794,0.0357293154837701,0.0377898022419026]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>p_val<\/th>\n      <th>avg_log2FC<\/th>\n      <th>pct.1<\/th>\n      <th>pct.2<\/th>\n      <th>p_val_adj<\/th>\n      <th>was_altered<\/th>\n      <th>AveExpr<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,7]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script><div class="sourceCode" id="cb69"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">de_result_negbinom</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">AveExpr</span>, y<span class="op">=</span><span class="va">avg_log2FC</span>, col<span class="op">=</span><span class="va">p_val_adj</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span>data<span class="op">=</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">de_result_negbinom</span>, <span class="va">was_altered</span><span class="op">)</span>, pch<span class="op">=</span><span class="fl">3</span>, col<span class="op">=</span><span class="st">'blue'</span>, size<span class="op">=</span><span class="fl">5</span> <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_colour_manual</a></span><span class="op">(</span>values<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'TRUE'</span><span class="op">=</span><span class="st">"red"</span>,<span class="st">'FALSE'</span><span class="op">=</span><span class="st">"black"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Negative Binomial Test"</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="scRNAseqInR_Doco_files/figure-html/unnamed-chunk-48-1.png" width="672"></div>
</div>
<div id="pseudobulk" class="section level2" number="12.6">
<h2>
<span class="header-section-number">12.6</span> Pseudobulk<a class="anchor" aria-label="anchor" href="#pseudobulk"><i class="fas fa-link"></i></a>
</h2>
<p>Pseudobulk analysis is an option where you have biological replicates. It is essentially pooling the individual cell counts and treating your expreiment like a bulk RNAseq.</p>
<p>First, you need to build a pseudobulk matrix - the <code><a href="https://satijalab.org/seurat/reference/AggregateExpression.html">AggregateExpression()</a></code> function can do this, once you set the ‘Idents’ of your seurat object to your grouping factor (here, sample instead of cluster).</p>
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Tools for bulk differential expression</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://bioinf.wehi.edu.au/limma">limma</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: package 'limma' was built under R version 4.1.3</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'limma'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:BiocGenerics':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     plotMA</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://bioinf.wehi.edu.au/edgeR">edgeR</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: package 'edgeR' was built under R version 4.1.1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'edgeR'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:SingleCellExperiment':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     cpm</span></span>
<span></span>
<span></span>
<span><span class="co"># Change idents to sample fro grouping.</span></span>
<span><span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/Idents.html">Idents</a></span><span class="op">(</span><span class="va">hcabm40k.mini.c0</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">hcabm40k.mini.c0</span><span class="op">$</span><span class="va">orig.ident</span></span>
<span></span>
<span></span>
<span><span class="va">pseudobulk_matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/AggregateExpression.html">AggregateExpression</a></span><span class="op">(</span> <span class="va">hcabm40k.mini.c0</span>,  slot <span class="op">=</span> <span class="st">'counts'</span>, assays<span class="op">=</span><span class="st">'RNA'</span> <span class="op">)</span><span class="op">[[</span><span class="st">'RNA'</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">pseudobulk_matrix</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="op">]</span></span>
<span><span class="co">#&gt;               MantonBM1 MantonBM2 MantonBM3 MantonBM4</span></span>
<span><span class="co">#&gt; FO538757.2           52        88       116        44</span></span>
<span><span class="co">#&gt; AP006222.2            4         4         4         8</span></span>
<span><span class="co">#&gt; RP11-206L10.9        20         4        36        20</span></span>
<span><span class="co">#&gt; NOC2L                64        72        52        60</span></span>
<span><span class="co">#&gt; ISG15               280       204        72       112</span></span>
<span><span class="co">#&gt;               MantonBM5 MantonBM6 MantonBM7 MantonBM8</span></span>
<span><span class="co">#&gt; FO538757.2            5         9        10        12</span></span>
<span><span class="co">#&gt; AP006222.2            0         5         3         2</span></span>
<span><span class="co">#&gt; RP11-206L10.9         8         2         3         1</span></span>
<span><span class="co">#&gt; NOC2L                 6        23        15        13</span></span>
<span><span class="co">#&gt; ISG15                18        55        50        36</span></span></code></pre></div>
<p>Now it looks like a bulk RNAseq experiment, so treat it like one.</p>
<div class="sourceCode" id="cb71"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dge</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/DGEList.html">DGEList</a></span><span class="op">(</span><span class="va">pseudobulk_matrix</span><span class="op">)</span></span>
<span><span class="va">dge</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/calcNormFactors.html">calcNormFactors</a></span><span class="op">(</span><span class="va">dge</span><span class="op">)</span></span>
<span></span>
<span><span class="va">condition</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">condition_lookup</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">pseudobulk_matrix</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">design</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html">model.matrix</a></span><span class="op">(</span><span class="op">~</span><span class="va">condition</span><span class="op">)</span></span>
<span><span class="va">vm</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/voom.html">voom</a></span><span class="op">(</span><span class="va">dge</span>, design <span class="op">=</span> <span class="va">design</span>, plot <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/lmFit.html">lmFit</a></span><span class="op">(</span><span class="va">vm</span>, design <span class="op">=</span> <span class="va">design</span><span class="op">)</span></span>
<span><span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html">eBayes</a></span><span class="op">(</span><span class="va">fit</span><span class="op">)</span></span>
<span><span class="va">de_result_pseudobulk</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html">topTable</a></span><span class="op">(</span><span class="va">fit</span>, n <span class="op">=</span> <span class="cn">Inf</span>, adjust.method <span class="op">=</span> <span class="st">"BH"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Removing intercept from test coefficients</span></span>
<span><span class="va">de_result_pseudobulk</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span><span class="op">(</span><span class="va">de_result_pseudobulk</span> , <span class="va">adj.P.Val</span><span class="op">)</span></span>
<span><span class="va">de_result_pseudobulk</span><span class="op">$</span><span class="va">was_altered</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">de_result_pseudobulk</span> <span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">altered_genes</span></span></code></pre></div>
<p>Look at the significantly differentially expressed genes:</p>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">DT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html">datatable</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">de_result_pseudobulk</span>,    <span class="va">adj.P.Val</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div id="htmlwidget-8433f375aeb6255a899c" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-8433f375aeb6255a899c">{"x":{"filter":"none","vertical":false,"data":[["SSU72","MIB2","FO538757.2","AURKAIP1","NOC2L","CCNL2","TNFRSF4","MRPL20","ISG15","CTSW","CPSF3L"],[1.82739857985341,2.21884180542971,3.05368793006276,1.87406229492983,2.25640207915933,2.00985486781816,2.68618784426133,1.74128751382085,2.07229191313832,-1.51944352793635,2.08318738510621],[8.22368206347954,5.995477411935,5.68324369086608,7.44806166173458,5.87362545270906,5.59400832966642,4.85785318344117,7.14036795661772,7.23175696091474,6.53375230251367,5.6728117668533],[8.69230448413311,8.52190131157026,8.06597361664774,7.5524361936642,7.28444847983461,6.41744035604559,5.91891963019649,5.72422192813309,5.48711892596761,-5.49769275663458,5.30880440933303],[2.86036261553234e-07,3.67880790112665e-07,7.33705998733036e-07,1.64679890113518e-06,2.54524812009268e-06,1.11200518863549e-05,2.71990693778898e-05,3.89293992961191e-05,6.06616475069649e-05,5.94640103426564e-05,8.50963697599627e-05],[0.00106133607947504,0.00106133607947504,0.00141116120422987,0.0023755074148875,0.00293721633058695,0.0106937832307113,0.0224198043300606,0.0280778292423259,0.0350017706115187,0.0350017706115187,0.0446369139559077],[6.66733411366858,5.6470506033109,4.79735742194687,5.05254647009648,4.18279685285552,2.99955073302207,1.87290006616061,2.32108740654014,1.93626352868609,1.88798542738999,1.42517599417348],[true,true,true,true,true,true,true,true,true,false,true]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>logFC<\/th>\n      <th>AveExpr<\/th>\n      <th>t<\/th>\n      <th>P.Value<\/th>\n      <th>adj.P.Val<\/th>\n      <th>B<\/th>\n      <th>was_altered<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script><p>And the stats for the altered genes:</p>
<div class="sourceCode" id="cb73"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">DT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html">datatable</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">de_result_pseudobulk</span>, <span class="va">was_altered</span> <span class="op">==</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div id="htmlwidget-4cb8b60b70fd71f3bf5a" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-4cb8b60b70fd71f3bf5a">{"x":{"filter":"none","vertical":false,"data":[["SSU72","MIB2","FO538757.2","AURKAIP1","NOC2L","CCNL2","TNFRSF4","MRPL20","ISG15","CPSF3L","CDK11B","SDF4","ATAD3B","UBE2J2","PUSL1","RP11-206L10.9","ACAP3","ATAD3A","AP006222.2","CPTP","TNFRSF18","RP11-345P4.9"],[1.82739857985341,2.21884180542971,3.05368793006276,1.87406229492983,2.25640207915933,2.00985486781816,2.68618784426133,1.74128751382085,2.07229191313832,2.08318738510621,1.69536089232766,1.66817853768678,2.4818967715993,1.57510105051403,1.53328867197325,2.51482754527967,1.46137529820723,1.76787386777385,1.23379003267208,1.04315803212992,1.43937829121061,1.40007865084643],[8.22368206347954,5.995477411935,5.68324369086608,7.44806166173458,5.87362545270906,5.59400832966642,4.85785318344117,7.14036795661772,7.23175696091474,5.6728117668533,5.86729295579645,6.35144160427787,3.44512119081969,6.03463887287738,3.90335650283279,3.88465100842493,4.36038576330553,3.70850399209292,2.79832482751051,3.64158512547261,3.43214453999052,3.70062401205564],[8.69230448413311,8.52190131157026,8.06597361664774,7.5524361936642,7.28444847983461,6.41744035604559,5.91891963019649,5.72422192813309,5.48711892596761,5.30880440933303,4.69429044836077,4.41610430641784,3.786880061361,3.46453095995809,3.3115071714009,3.22109402082098,2.73527829316228,2.64857516878448,1.9872264791775,1.83119190738041,1.57865691002138,1.42599811157796],[2.86036261553234e-07,3.67880790112665e-07,7.33705998733036e-07,1.64679890113518e-06,2.54524812009268e-06,1.11200518863549e-05,2.71990693778898e-05,3.89293992961191e-05,6.06616475069649e-05,8.50963697599627e-05,0.000281604943590021,0.000490843976682095,0.00176629608832836,0.00342895170027388,0.004699197629499,0.00565966294965288,0.0152424103096016,0.0181439084383963,0.0653128034034893,0.086836036849633,0.135091552065387,0.174176618095009],[0.00106133607947504,0.00106133607947504,0.00141116120422987,0.0023755074148875,0.00293721633058695,0.0106937832307113,0.0224198043300606,0.0280778292423259,0.0350017706115187,0.0446369139559077,0.132566904644541,0.188811316363712,0.536396233139717,0.73277967816964,0.821647585521492,0.933035863414203,0.997887502406555,0.997887502406555,0.997887502406555,0.997887502406555,0.997887502406555,0.997887502406555],[6.66733411366858,5.6470506033109,4.79735742194687,5.05254647009648,4.18279685285552,2.99955073302207,1.87290006616061,2.32108740654014,1.93626352868609,1.42517599417348,0.490383172374314,0.0537051707433776,-1.52858994725577,-1.64433324517802,-2.04781450767965,-2.22812983546374,-2.87920934416095,-3.04649462051618,-3.9280798768479,-4.13957382059417,-4.41616654444229,-4.61353678072249],[true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>logFC<\/th>\n      <th>AveExpr<\/th>\n      <th>t<\/th>\n      <th>P.Value<\/th>\n      <th>adj.P.Val<\/th>\n      <th>B<\/th>\n      <th>was_altered<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script><div class="sourceCode" id="cb74"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">de_result_pseudobulk</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">AveExpr</span>, y<span class="op">=</span><span class="va">logFC</span>, col<span class="op">=</span><span class="va">adj.P.Val</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span>data<span class="op">=</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">de_result_pseudobulk</span>, <span class="va">was_altered</span><span class="op">)</span>, pch<span class="op">=</span><span class="fl">3</span>, col<span class="op">=</span><span class="st">'blue'</span>, size<span class="op">=</span><span class="fl">5</span> <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_colour_manual</a></span><span class="op">(</span>values<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'TRUE'</span><span class="op">=</span><span class="st">"red"</span>,<span class="st">'FALSE'</span><span class="op">=</span><span class="st">"black"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Pseudobulk"</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="scRNAseqInR_Doco_files/figure-html/unnamed-chunk-53-1.png" width="672"></div>
<div id="challenge-what-method" class="section level4 unnumbered rmdtip">
<h4>Challenge: What method?<a class="anchor" aria-label="anchor" href="#challenge-what-method"><i class="fas fa-link"></i></a>
</h4>
<p>These methods give different results. How could and decide which to use? How could you check an individual gene?</p>

</div>
</div>
</div>



  <div class="chapter-nav">
<div class="prev"><a href="singler.html"><span class="header-section-number">11</span> SingleR</a></div>
<div class="next"><a href="finishing-up.html"><span class="header-section-number">13</span> Finishing up</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#de"><span class="header-section-number">12</span> Differential Expression</a></li>
<li><a class="nav-link" href="#prepared-data"><span class="header-section-number">12.1</span> Prepared data</a></li>
<li><a class="nav-link" href="#load-data"><span class="header-section-number">12.2</span> Load Data</a></li>
<li><a class="nav-link" href="#prefiltering"><span class="header-section-number">12.3</span> Prefiltering</a></li>
<li><a class="nav-link" href="#default-wilcox-test"><span class="header-section-number">12.4</span> Default Wilcox test</a></li>
<li><a class="nav-link" href="#seurat-negative-binomial"><span class="header-section-number">12.5</span> Seurat Negative binomial</a></li>
<li><a class="nav-link" href="#pseudobulk"><span class="header-section-number">12.6</span> Pseudobulk</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/rstudio/bookdown-demo/blob/master/04-10-de.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/rstudio/bookdown-demo/edit/master/04-10-de.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>scRNAseq Analysis in R with Seurat</strong>" was written by . It was last built on 2022-08-11.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
